### DISCUSSION TITLE
How to search without space

### DISCUSSION THREAD

**Discussion by matart15** [supabase/supabase#18906] [Questions] [answered]
Japanese sentences does not include space between words.

for example my manual title is something like this: `接続マニュアル`. which is two word

```ts
 const supabaseQuery = supabase
    .from("manuals")
    .select("*", { count: "exact" })
    .textSearch("title", `'${keyword}'`, {
      type: "websearch",
      // config: "japanese",  <---- looks like anything except english gives error 
    });
```

if i search with whole word `接続マニュアル` it returns result

But if i search by first word it returns nothing `接続`. I assume it does not support subsstring search.

How can I achieve this?

**Comment by charislam** [comment, marked as answer]
Have you taken a look at this extension? https://supabase.com/docs/guides/database/extensions/pgroonga

I haven't tried it myself, but it sounds like it may do what you are looking for

**Reply by matart15** [reply]
Thank you.

I will try it.
